### PR TITLE
Updated for Jekyll 4

### PR DESCRIPTION
**Pull request by SDBrett**
#### Feedback required on forking of jekyll-github-metadata plugin before approval.

The version of Jekyll used to build the site has been updated from 3.8.5 to 4.0.

makefile updated to use Jekyll 4 container image
gemfile updated to provide control over plugin versions
gemfile.lock updated for new gem versions
.gitignore updated to ignore jekyll cache directory created during site testing. Removed duplication of .DS_Store

Signed-off-by: Brett Johnson <brett@sdbrett.com>

### CODE REVIEW

**Comment by SDBrett** [comment]
This will need additional review work after PR #1838 has been merged into master.

**Comment by jpeach** [comment]
@SDBrett What's the issue with `github-metadata`?

**Comment by SDBrett** [comment]
> @SDBrett What's the issue with `github-metadata`?

That was a mistake, couldn't find a use for it until after I raised the PR.

**Comment by SDBrett** [comment]
This should be put on hold until after #1838 is merged as there will need to be additional testing for TOC comparability when changing from red carpet to kramdown

**Comment by jpeach** [comment]
@jonasrosland Can you please take a look?

**Comment by jonasrosland** [comment]
Will do, will chat with Brett later today about this. Looks good from a quick glance! Glad we have someone with much more Jekyll knowledge :)

File: .gitignore [hunk 1]
```diff
@@ -3,7 +3,6 @@
 vendor/
 .vs/
 .idea/
-.DS_Store
```
Comment (jonasrosland) [inline .gitignore]: Why remove this?

File: site/Gemfile [hunk 2]
```diff
@@ -1,5 +1,12 @@
...
+    gem 'jekyll', '~> 4'
+    gem 'jekyll-seo-tag', '~> 2.6', '>= 2.6.1' # Automatically create SEO tags
+    gem 'jekyll-sitemap', '~> 1.3', '>= 1.3.1' # Create a sitemap using the official Jekyll sitemap gem
+    gem 'jekyll-feed', '~> 0.12.1' # Create an Atom feed using the official Jekyll feed gem
+    gem 'jekyll-relative-links', '~> 0.6.1' # Used to auto generate md links to html links
+    gem 'jekyll-paginate', '~> 1.1' # pagination object for collections (e.g. posts)
+    gem "jekyll-github-metadata", github: "jekyll/github-metadata" # TODO fork repo and update github link reference
+    gem 'jekyll-optional-front-matter', '~> 0.3.2' # Parse Markdown files that do not have front-matter callouts
+    gem 'jekyll-titles-from-headings', '~> 0.5.3' # pull the page title from the first Markdown heading when none is specified.)
+end
```
Comment (jonasrosland) [inline site/Gemfile]: Versioning the gems = A+, thank you.

File: site/_config.yml [hunk 3]
```diff
@@ -10,7 +10,7 @@ twitter:
   card: summary
 vm_logo: vm-logo.png
 gh_repo: https://github.com/projectcontour/contour
-markdown: redcarpet
+markdown: kramdown
```
Comment (jonasrosland) [inline site/_config.yml]: Is there anything we're missing out on by going with kramdown vs redcarpet? I just want to make sure we've covered that.

File: site/Gemfile.lock [hunk 4]
```diff
@@ -1,252 +1,102 @@
+GIT
+  remote: https://github.com/jekyll/github-metadata.git
+  revision: 9353743dc9ad6de2f522923d57eb8eb00f87abf6
+  specs:
+    jekyll-github-metadata (2.12.1)
+      jekyll (>= 3.4, < 5.0)
+      octokit (~> 4.0, != 4.4.0)
+
```
Comment (jonasrosland) [inline site/Gemfile.lock]: We can't include this as a regular gem yet, is that correct? What's the course of action when the gem is available with support for Jekyll 4?

**Comment by jonasrosland** [comment]
Can you please see if there are any changes that we need to make from the descriptions in the build results?

```8:24:07 PM: Post-install message from i18n:
8:24:07 PM: HEADS UP! i18n 1.1 changed fallbacks to exclude default locale.
8:24:07 PM: But that may break your application.
8:24:07 PM: Please check your Rails app for 'config.i18n.fallbacks = true'.
8:24:07 PM: If you're using I18n (>= 1.1.0) and Rails (< 5.2.2), this should be
8:24:07 PM: 'config.i18n.fallbacks = [I18n.default_locale]'.
8:24:07 PM: If not, fallbacks will be broken in your app by I18n 1.1.x.
8:24:07 PM: For more info see:
8:24:07 PM: https://github.com/svenfuchs/i18n/releases/tag/v1.1.0
8:24:07 PM: Post-install message from jekyll:
8:24:07 PM: -------------------------------------------------------------------------------------
8:24:07 PM: Jekyll 4.0 comes with some major changes, notably:
8:24:07 PM:   * Our `link` tag now comes with the `relative_url` filter incorporated into it.
8:24:07 PM:     You should no longer prepend `{{ site.baseurl }}` to `{% link foo.md %}`
8:24:07 PM:     For further details: https://github.com/jekyll/jekyll/pull/6727
8:24:07 PM:   * Our `post_url` tag now comes with the `relative_url` filter incorporated into it.
8:24:07 PM:     You shouldn't prepend `{{ site.baseurl }}` to `{% post_url 2019-03-27-hello %}`
8:24:07 PM:     For further details: https://github.com/jekyll/jekyll/pull/7589
8:24:07 PM:   * Support for deprecated configuration options has been removed. We will no longer
8:24:07 PM:     output a warning and gracefully assign their values to the newer counterparts
8:24:07 PM:     internally.
8:24:07 PM: -------------------------------------------------------------------------------------
```

**Comment by jonasrosland** [comment]
Can you please rebase this as well, I can't get the preview to build correctly?

File: site/Gemfile.lock (hunk 4 above)
Comment (SDBrett) [reply site/Gemfile.lock]: That's correct, the Gemfile will be updated when a gem is released which supports Jekyll 4 and PR .

File: site/_config.yml (hunk 3 above)
Comment (SDBrett) [reply site/_config.yml]: I'm in the process of validating it. now that the docs update has been merged. There is no support for Redcarpet in Jekyll 4.

File: site/Gemfile (hunk 2 above)
Comment (SDBrett) [reply site/Gemfile]: :)

File: .gitignore (hunk 1 above)
Comment (SDBrett) [reply .gitignore]: Duplicate entry, line 6 and 14

**Comment by SDBrett** [comment]
Working through Netlify build failures due to Bundler version

**Comment by SDBrett** [comment]
> Working through Netlify build failures due to Bundler version

Downgraded bundler to 1.7.3 to resolve issue.

Reference: https://github.com/netlify/build-image/issues/296

**Comment by SDBrett** [comment]
> 8:24:07 PM: Jekyll 4.0 comes with some major changes, notably:
> 8:24:07 PM:   * Our `link` tag now comes with the `relative_url` filter incorporated into it.
> 8:24:07 PM:     You should no longer prepend `{{ site.baseurl }}` to `{% link foo.md %}`
> 8:24:07 PM:     For further details: https://github.com/jekyll/jekyll/pull/6727
> 8:24:07 PM:   * Our `post_url` tag now comes with the `relative_url` filter incorporated into it.
> 8:24:07 PM:     You shouldn't prepend `{{ site.baseurl }}` to `{% post_url 2019-03-27-hello %}`
> 8:24:07 PM:     For further details: https://github.com/jekyll/jekyll/pull/7589
> 8:24:07 PM:   * Support for deprecated configuration options has been removed. We will no longer
> 8:24:07 PM:     output a warning and gracefully assign their values to the newer counterparts
> 8:24:07 PM:     internally.

Those links are not used, this is a non-issue. Just a warning when Jekyll 4 is executed.

**Comment by SDBrett** [comment]
@jonasrosland Have I answered all your questions for the PR?

**Comment by jonasrosland** [comment]
> @jonasrosland Have I answered all your questions for the PR?

I think so! I see that Netlify now supports Bundler 2.0.2, do you want to update this PR to include that?

**Comment by stevesloka** [comment]
Is this ready to go @SDBrett? Curious about the last comment from @jonasrosland regarding bundler.

**Comment by SDBrett** [comment]
> Is this ready to go @SDBrett? Curious about the last comment from @jonasrosland regarding bundler.

I'll update bundler in the build, rebase and validate again. 
Likely get to that tomorrow.

**Comment by SDBrett** [comment]
> Is this ready to go @SDBrett? Curious about the last comment from @jonasrosland regarding bundler.

I'll update bundler in the build, rebase and validate again. 
Likely get to that tomorrow.

**Comment by SDBrett** [comment]
> Is this ready to go @SDBrett? Curious about the last comment from @jonasrosland regarding bundler.

I'll update bundler in the build, rebase and validate again. 
Likely get to that tomorrow.

**Comment by SDBrett** [comment]
> Is this ready to go @SDBrett? Curious about the last comment from @jonasrosland regarding bundler.

I'll update bundler in the build, rebase and validate again. 
Likely get to that tomorrow.

**Pull request merged**